### PR TITLE
feat: increase CSS specificity of components

### PR DESCRIPTION
**Pull request by snitin315**
[
JIRA Ticket](https://razorpay.atlassian.net/browse/FC-2429?atlOrigin=eyJpIjoiM2MyODZiY2M2MWJlNGM3YThkY2Q4MGE4OGVjZjBiMTciLCJwIjoiaiJ9)

### Summary by CodeRabbit

- New Feature: Increased CSS specificity of components in BladeProvider, Box, Button, and styledProps modules to improve styling capabilities.
- Refactor: Added import statement for `StyleSheetManager` from the `styled-components` library in BladeProvider module.
- Refactor: Created a new function `createStylisCSSHigherSpecificityPlugin` in BladeProvider module to increase CSS specificity by repeating the first selector multiple times.
- Style: Updated CSS class names in Box and styledProps modules to increase specificity.
- Chore: Added a new `namespace` property with the value `'razorpay-blade'` to .babelrc.js configuration in Blade module.

### CODE REVIEW

File: packages/examples/basic/yarn.lock [hunk 1]
```diff
@@ -9458,4 +9458,4 @@ yargs@^16.2.0:
 yocto-queue@^0.1.0:
   version "0.1.0"
   resolved "https://registry.yarnpkg.com/yocto-queue/-/yocto-queue-0.1.0.tgz#0294eb3dee05028d31ee1a5fa2c556a6aaf10a1b"
-  integrity sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==
+  integrity sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==
```
Comment (kamleshchandnani) [inline packages/examples/basic/yarn.lock:9461]: some issues here? maybe try running yarn.lock again?

File: packages/blade/src/components/BladeProvider/stylisCSSHigherSpecificity.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import type { StylisPlugin } from 'styled-components';
+
```
Comment (kamleshchandnani) [inline packages/blade/src/components/BladeProvider/stylisCSSHigherSpecificity.ts:2]: did you referf this plugin? can you check how is this different than ours in terms of behaviour?
https://github.com/sanusart/stylis-plugin-extra-class-names-specifity/blob/main/index.js

Comment (snitin315) [reply packages/blade/src/components/BladeProvider/stylisCSSHigherSpecificity.ts:2]: @kamleshchandnani this seems to be doing the same thing, this seems to have some additional checks on [`@keyframe`](https://github.com/sanusart/stylis-plugin-extra-class-names-specifity/blob/2cc91d4cd2253fdfeb2e9e27d8065d4fc632f9a9/test.js#L18-L20) and [`nested/chained` styles](https://github.com/sanusart/stylis-plugin-extra-class-names-specifity/blob/2cc91d4cd2253fdfeb2e9e27d8065d4fc632f9a9/test.js#L22).

Also, this plugin is using stylis v4, hence the syntax is different. [styled-component v6 or later](https://styled-components.com/releases#v6.0.0) is required to use stylis v4.

File: .changeset/tiny-pugs-sort.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+"@razorpay/blade": minor
+---
+
+feat: increase the CSS specificity of all components
+
+> [!WARNING]
+>
+> **We've enhanced the specificity of styles for our components. If you've defined styles in a separate CSS file targeting blade components, please note that they may no longer take effect as expected.
```
[the remark this answers is not in the document]
Comment (snitin315) [reply .changeset/tiny-pugs-sort.md:9]:
```suggestion
> **We've enhanced the specificity of styles for Blade components. If you've defined styles in separate CSS at a root level targeting blade components, please note that they may no longer take effect as expected.
```

**Pull request merged**